src/lib.rs

```diff
@@ -1073,31 +1089,63 @@ impl AssociatedType {
     }
 }
 
+// ===== impl Field =====
+
+impl Field {
+    /// Return a field definition with the provided name and type
+    pub fn new<T>(name: &str, ty: T) -> Self
+    where T: Into<Type>,
+    {
+        Field {
+            name: name.into(),
+            ty: ty.into(),
+            documentation: String::new(),
+            annotation: String::new(),
+        }
+    }
+
+    /// Set field's documentation.
+    pub fn with_documentation(&mut self, documentation: &str) -> &mut Self {
+        self.documentation = documentation.into();
+        self
+    }
+
+    /// Set field's annotation.
+    pub fn with_annotation(&mut self, annotation: &str) -> &mut Self {
```

Comment:
Could you name this `annotation`? I'm avoiding `with_` prefixes.

Reply:
Done in 5aafce7.